From d55e688a4fab8307b0798b71be82cac3a23886f7 Mon Sep 17 00:00:00 2001
From: Shunqian Zheng <zhengsq@rock-chips.com>
Date: Thu, 21 Jun 2018 17:16:53 +0800
Subject: [PATCH] ARM: dts: rockchip: override codec i2s_clk in rk3128.dtsi

Inside rk3128, the codec connects to i2s_8ch which is different
from the rk3126. This changes the i2s_clk to SCLK_I2S0, the mclk
of i2s_8ch, for rk3128.

Change-Id: I204e46dc2cf2c05bc85ef233b3139af94445ac90
Signed-off-by: Shunqian Zheng <zhengsq@rock-chips.com>
---
 arch/arm/boot/dts/rk3128.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/boot/dts/rk3128.dtsi b/arch/arm/boot/dts/rk3128.dtsi
index 493fc8145272..a116a71c39a4 100644
--- a/arch/arm/boot/dts/rk3128.dtsi
+++ b/arch/arm/boot/dts/rk3128.dtsi
@@ -72,6 +72,15 @@
 	};
 };
 
+&codec {
+	/*
+	 * Override the i2s_clk since codec connects to i2s_8ch in rk3128,
+	 * which is different from rk3126.
+	 */
+	clock-names = "g_pclk_acodec", "i2s_clk";
+	clocks = <&cru PCLK_ACODEC>, <&cru SCLK_I2S0>;
+};
+
 &pd_vio {
 	pm_qos = <&qos_rga>,
 		 <&qos_ebc>,
-- 
2.35.3

